ui/src/app/pages/login/account-creation/account-creation.spec.tsx

```diff
@@ -103,128 +97,116 @@ test.each([
   'no+plus',
   'ælfred',
   'money$man',
-])('should mark username %s as invalid', (username) => {
-  const wrapper = component();
-  expect(wrapper.exists('#username')).toBeTruthy();
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
-  wrapper
-    .find('input#username')
-    .simulate('change', { target: { value: username } });
-  expect(wrapper.exists('#usernameError')).toBeTruthy();
+])('should mark username %s as invalid', async (username) => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('#username')).not.toBeNull();
+  expect(container.querySelector('#usernameError')).toBeNull();
+  await user.type(container.querySelector('input#username'), username);
+  expect(container.querySelector('#usernameError')).not.toBeNull();
 });
 
-it('should handle username validity long but has mismatch at end', () => {
-  const wrapper = component();
-  expect(wrapper.exists('#username')).toBeTruthy();
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
+it('should handle username validity long but has mismatch at end', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('#username')).not.toBeNull();
+  expect(container.querySelector('#usernameError')).toBeNull();
   // if username is long (not too long) but has a mismatch at end
   let testInput = fp.repeat(50, 'abc');
   testInput = testInput + ' abc';
-  wrapper
-    .find('input#username')
-    .simulate('change', { target: { value: testInput } });
-  expect(wrapper.exists('#usernameError')).toBeTruthy();
+  await user.type(container.querySelector('input#username'), testInput);
+  expect(container.querySelector('#usernameError')).not.toBeNull();
 });
 
-it('should handle username validity length less than 3 characters', () => {
-  const wrapper = component();
-  expect(wrapper.exists('#username')).toBeTruthy();
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
-  wrapper.find('input#username').simulate('change', { target: { value: 'a' } });
-  expect(wrapper.exists('#usernameError')).toBeTruthy();
+it('should handle username validity length less than 3 characters', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('#username')).not.toBeNull();
+  expect(container.querySelector('#usernameError')).toBeNull();
+  await user.type(container.querySelector('input#username'), 'a');
+  expect(container.querySelector('#usernameError')).not.toBeNull();
```

Comment:
💯, I hear you. In the context of "I'm converting this since we're trying to convert tests _as part of other work_" I think it's expected and acceptable to do the minimum possible conversion and move on.